packages/core/src/app/cart/mapToOrderSummarySubtotalsProps.ts

```diff
@@ -14,15 +14,27 @@ export default function mapToOrderSummarySubtotalsProps({
     coupons,
     taxes,
     fees,
-}: Checkout): OrderSummarySubtotalsProps {
+}: Checkout,
+isShippingDiscountDisplayEnabled: boolean,
+): OrderSummarySubtotalsProps {
+    const allConsignmentsHaveSelectedShippingOption = hasSelectedShippingOptions(consignments);
+    const shippingAmount = isShippingDiscountDisplayEnabled
+        ? allConsignmentsHaveSelectedShippingOption
+            ? getShippingCostAfterAutomaticDiscount(shippingCostBeforeDiscount, consignments)
+            : undefined
+        : allConsignmentsHaveSelectedShippingOption
+            ? shippingCostBeforeDiscount
+            : undefined;
```

Comment:
Can you create a function for this calculation?

Reply:
Yes, the code may look duplicated, but I prefer to keep it separate. This makes it easier to remove the old code when cleaning up the experiment, reducing the risk of mistakes. WDYT?